include/http/request_handler.js

```diff
@@ -128,7 +53,625 @@ module.exports = function RequestHandlerModule(pb) {
      * @property PUBLIC_ROUTE_PREFIXES
      * @type {Array}
      */
-    var PUBLIC_ROUTE_PREFIXES = ['/js/', '/css/', '/fonts/', '/img/', '/localization/', '/favicon.ico', '/docs/', '/bower_components/'];
+    const PUBLIC_ROUTE_PREFIXES = ['/js/', '/css/', '/fonts/', '/img/', '/localization/', '/favicon.ico', '/docs/', '/bower_components/'];
+
+    const GLOBAL_SITE = pb.SiteService.GLOBAL_SITE;
+
+    class RequestHandler {
+        constructor(req, resp) {
+            /**
+             * @property req
+             * @type {Request}
+             */
+            this.req = req;
+            /**
+             * @property resp
+             * @type {Response}
+             */
+            this.resp = resp;
+            /**
+             * @property errorCount
+             * @type {number}
+             */
+            this.errorCount = 0;
+        }
+        /**
+             * Derives the locale and localization instance.
+             * @method deriveLocalization
+             * @param {Object} context
+             * @param {Object} [context.session]
+             * @param {String} [context.routeLocalization]
+             */
+        deriveLocalization(context) {
+            var opts = {};
```

Comment:
Hey, so this is really minor, but just a thought:
It looks like you're pushing this.req.headers[pb.Localization.ACCEPT_LANG_HEADER] instead of including it in sources  at definition time because the order of these items matters; 
if that's the case then it might be cool to derive the set of sources from an array of their paths, in the order of their priority:
```

let sources = ['context.routeLocalization', 'context.session.locale', `this.req.headers.${ACCEPT_LANG_HEADER}`, 'this.siteObj.defaultLocale'].map(_.get).filter(val => {return !!val;});

```

Reply:
Agreed, and that is what https://github.com/cbdr/CMSPencilblue/pull/1452/files#diff-8f3f7c283e626c7dcc9241d14b88a10dR14 does